src/custom/api/coingecko/index.ts

```diff
@@ -93,10 +96,10 @@ export async function getUSDPriceQuote(params: CoinGeckoUsdPriceParams): Promise
 }
 
 export function useGetCoingeckoUsdPrice(params: Partial<CoinGeckoUsdPriceParams>, options = SWR_OPTIONS) {
-  const { chainId, tokenAddress } = params
+  const { chainId, tokenAddress, isNative } = params
 
   return useSWR<PriceInformation | null>(
-    ['getUSDPriceQuote', chainId, tokenAddress],
+    ['getUSDPriceQuote', chainId, tokenAddress, isNative],
```

Comment:
It looks like you are ignoring the `isNative` here.  This is where I was kind of expecting to see some conditional logic